reproman/resource/session.py

```diff
@@ -523,6 +521,40 @@ def _parse_envvars_output(self, out):
             output[split[0]] = split[1]
         return output
 
+    def _prefix_command(self, command, env=None, cwd=None, with_shell=True):
+        """Wrap the command in a shell call with ENV vars and CWD prefixed
+        statment to command. Will pass through the command unchanged if env
+        and cwd are None
+
+        Parameters
+        ----------
+        command : string or list
+            shell command to be run
+        env : dict
+            ENV vars to prefix to command
+        cwd : string
+            working directory to cd into befoe the command is executed
+        with_shell : boolean
+            if True, a "/bin/sh -c" call will wrap the cwd, env and command
+
+        Returns
+        -------
+        string
```

Comment:
But not if `env` and `cwd` are false, right?  I suspect it'd be cleanest to always give back a string.  Plus I think the caller would expect `with_shell` to have an effect even if `env` and `cwd` are false.

Reply:
The _prefix_command method simply returns the original string if env and cwd are false.

Right on the with_shell, it should add the /bin/sh invokation even without env or cwd

Reply:
> The _prefix_command method simply returns the original string if env and cwd are false.

But my point is that the command can come in as a list, I think, so you'd be returning a list.

Reply:
Yeah, I figured that out after I commented. There are a couple of problems. I have sorted it out now but have triggered some test fails that need to be tracked down and fixed